1_fetch/src/sb_fetch.py

```diff
@@ -0,0 +1,70 @@
+# Download ScienceBase items by item id, and optionally by file name
+
+import os
+from sciencebasepy import SbSession
+
+def sb_get(sb_session, item_id, sb_data_file=None, destination_dir='.'):
+    """
+    Download an item from ScienceBase by ID
+
+    :param sb_session: Active SbSession
+    :param item_id: ScienceBase ID of item to download
+    :param sb_data_file: Name of file to download. If None, download all files. (Default value = None)
+    :param destination_dir: directory to save to (Default value = '.')
+    :returns: ScienceBase JSON response
+
+    """
+    if not (sb_session.ping()['result'] == 'OK'):
+        raise ConnectionError('ScienceBase ping unsuccessful')
+    # If data are public, no need to log in
+    # if not sb_session.is_logged_in():
+        # sb_session.login()
+
+    item_json = sb_session.get_item(item_id)
+    if sb_data_file is None:
+        response = sb_session.get_item_files(item_json, destination=destination_dir)
+    else:
+        all_file_info = sb_session.get_item_file_info(item_json)
+        file_found = False
+        for file_info in all_file_info:
+            if file_info['name'] == sb_data_file:
+                file_found = True
+                response = sb_session.download_file(file_info['url'], file_info['name'], destination_dir)
+        if not file_found:
+            raise FileNotFoundError(f'{sb_data_file} not found on ScienceBase')
+
+    return response
+
+
+def main(item_id, filepath=None):
+    """
+    Download a ScienceBase item
+
+    :param item_id: ScienceBase ID of item to download
+    :param filepath: Name of file to download, and path to download to.
+        If None, download all files. (Default value = None)
+    :returns: ScienceBase JSON response
+
+    """
+    # for a default None filepath, download all files in the item
+    if filepath is None:
+        filename = None
+        destination_dir='.'
+        item_str = item_id
+    else:
+        filename = os.path.basename(filepath)
+        destination_dir = os.path.dirname(filepath)
+        if not os.path.exists(destination_dir):
+            os.makedirs(destination_dir)
+        if filename == '':
```

Comment:
Similarly to my comment on line 70 - when would this be the case, given the current setup? if you provided this, with just a space or nothing after the hash?
```
mntoha:

  metadata:
      sb_id: 5e5c1c1ce4b01d50924f27e7
      files:
         -  
  obs:
      sb_id: 5e5d0b68e4b01d50924f2b32
      files:
          - temperature_observations.zip
```
And if that is simply a space (or an empty string) for the `['metadata']['files']` value in the `mntoha` dictionary, then does `snakemake` match `''`  for `{file}` in the wildcard using the `"1_fetch/out/{file_category}_mntoha/{file}"` pattern?

Reply:
Right, as above, that logic won't get triggered by `snakemake` either.

Reply:
If we wanted to download all the files in one SB item... we could rewrite `main` in `sb_fetch.py`, and probably include a boolean in Snakemake.params that signifies whether we want the single files or all files in the SB item. Do you think we'll want that functionality at some point?

Reply:
(I tried a few iterations of None/empty string for `['metadata']['files']` and got errors)

Reply:
I think that leaving as is (having to specify the files) is fine for now, given the current use case, which is the mntoha data. If we switch to a new data sources, that might either 1) motivate adding the option to download all files from a given SB id or 2) require changes to this framework anyhow (e.g., a total hypothetical, but maybe we decide to store data on caldera instead and pull with globus sdk)